File: client/Model/recipe.py
from Model.recipe_ingredient import *


class Recipe(object):
    """A Recipe Class
    
    Attributes:
        name: A string representing the recipes's name.
        image: A string represting recipe's image's url.
        ingredients: A list of ingredient objects.
        directions: A string representing how to make the recipe.
    """

    def __init__(self, name='', image=None, ingredients=[RecipeIngredient()], directions='', tips=''):
        """Return a Recipe object whose name is *name* and starting
        directions is *directions*."""
        self.name = name
        self.image = image
        self.ingredients = ingredients
        self.directions = directions
        self.tips = tips

File: client/Webpack/src/addrecipe/actions/recipe.js
export const EDIT_RECIPE_NAME = 'EDIT_RECIPE_NAME';
export const editRecipeName = (name) => {
    return {
        type: EDIT_RECIPE_NAME,
        name
    }
};

export const EDIT_DIRECTIONS = 'EDIT_DIRECTIONS';
export const editDirections = (directions) => {
    return {
        type: EDIT_DIRECTIONS,
        directions
    }
};

export const EDIT_TIPS = 'EDIT_TIPS';
export const editTips = (tips) => {
    return {
        type: EDIT_TIPS,
        tips
    }
};